code/game/objects/effects/spawners/windowspawner.dm

```diff
@@ -20,35 +21,34 @@
 			for(var/obj/effect/spawner/window/WS in get_step(src,cdir))
 				cdir = null
 				break
-			if(!cdir)	continue
-			var/obj/structure/window/WI = new windowtospawn(get_turf(src))
+			if(!cdir)
+				continue
+			var/obj/structure/window/WI = new window_to_spawn_regular(get_turf(src))
 			WI.dir = cdir
 	else
-		var/obj/structure/window/W = new windowtospawn(get_turf(src))
-		W.dir = SOUTHWEST
+		new window_to_spawn_full(get_turf(src))
 
 	if(useGrille)
 		new /obj/structure/grille(get_turf(src))
 
-	src.air_update_turf(1) //atmos can pass otherwise
+	air_update_turf(TRUE) //atmos can pass otherwise
 	// Give some time for nearby window spawners to initialize
-	spawn(10)
-		qdel(src)
-	// why is this line a no-op
-	// QDEL_IN(src, 10)
+	QDEL_IN(src, 10)
```

Comment:
This can probably be `QDEL_IN(src, 0)` since `SStimer` doesnt actually fire until the MC has fully started, and these are setup during the `SSatoms` portion

Reply:
I just did 'return INITIALIZE_HINT_QDEL` instead. I don't think this needs any spawn like mechanism anymore.